src/main.cpp

```diff
@@ -4643,6 +4644,25 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     }
 
 
+    // Disconnect existing peer connection when:
+    // 1. Minimum peer version is less than Overwinter version
+    // 2. The version message has been received from a peer
+    // 3. The peer's version is pre-Overwinter
+    // 4. Overwinter is active
+    else if (
+        MIN_PEER_PROTO_VERSION < OVERWINTER_PROTO_VERSION &&
+        pfrom->nVersion != 0 &&
```

Comment:
This condition fragment cannot be false, as there is a `pfrom->nVersion == 0` check above here.